core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -577,23 +577,17 @@ class KafkaConfig private(doLog: Boolean, val props: util.Map[_, _])
       if (processRoles.isEmpty) {
         throw new ConfigException(s"The new '${GroupType.CONSUMER}' rebalance protocol is only supported in KRaft cluster.")
       }
-      warn(s"The new '${GroupType.CONSUMER}' rebalance protocol is enabled along with the new group coordinator. " +
-        "This is part of the preview of KIP-848 and MUST NOT be used in production.")
     }
     if (protocols.contains(GroupType.SHARE)) {
-      // The CONSUMER protocol enables the new group coordinator, and that's a prerequisite for share groups.
-      if (!protocols.contains(GroupType.CONSUMER)) {
```

Comment:
This line now should check whether the new GC is enabled. By default, that will be. If someone turns it off, they can't have share groups either.

Reply:
That's fair. Let me add it.